zjit/src/hir.rs

```diff
@@ -6548,6 +6604,116 @@ mod opt_tests {
         ");
     }
 
+    #[test]
+    fn test_optimize_range_fixnum_inclusive_literals() {
+        eval("
+            def test()
+              (1..2)
+            end
+            test; test
+        ");
+        assert_snapshot!(hir_string("test"), @r"
+        fn test@<compiled>:3:
+        bb0(v0:BasicObject):
+          v2:RangeExact[VALUE(0x1000)] = Const Value(VALUE(0x1000))
+          CheckInterrupts
+          Return v2
+        ");
+    }
+
+    #[test]
+    fn test_optimize_range_fixnum_exclusive_literals() {
+        eval("
+            def test()
+              (1...2)
```

Comment:
You may have noticed that this doesn't produce a New range instruction. For that you probably need to use a local variable (makes it harder for the bytecode compiler to reason about, but the JIT sees through it). If you have time, please rewrite these tests.